src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2334,6 +2376,32 @@ void MarkInterfaceImplementations (TypeDefinition type)
 				if (ShouldMarkInterfaceImplementation (type, iface))
 					MarkInterfaceImplementation (iface, new MessageOrigin (type));
 			}
+
+			bool ShouldMarkInterfaceImplementation (TypeDefinition type, InterfaceImplementation iface)
```

Comment:
This is moved to a local method because it assumes that 'type' is marked as instantiated, so we wouldn't want to call it elsewhere.